### ISSUE TITLE
Build breaks on armv7, i386

### ISSUE THREAD

**Issue by yurivict** [sadko4u/lsp-plugins#266]
On armv7:
```
<inline asm>:21:26: note: instantiated into assembly here
        vext.32     q4, q4, q4, $2
                                ^
In file included from main/arm/neon-d32.cpp:51:
/wrkdirs/usr/ports/audio/lsp-plugins-lv2/work/lsp-plugins/modules/lsp-dsp-lib/include/private/dsp/arch/arm/neon-d32/copy.h:371:17: error: operand must be an immediate in the range [0,15]
                __ASM_EMIT("vext.32     q4, q4, q4, $2")
                ^
```

It also breaks on i386 with the error: inline assembly requires more registers than available.

Version: 1.2.1
clang-13
FreeBSD 13.1

**Comment by sadko4u** [comment]
For me it's not possible to perform build for ARM at this moment. Any ways to reproduce?

**Comment by yurivict** [comment]
Unfortunately I don't have access to such systems.
The error occurred on central build FreeBSD server.

**Comment by sadko4u** [comment]
For ARM that seems that inline assembly doesn't like the numbers like `$2` and strictly requires `#2` instead.
I think this is simple to fix.

**Comment by sadko4u** [comment]
For building on i386: reproduced the case.
It seems that clang is behaving a bit different when allocating registers.
Consider the following example:

```
        void lr_to_ms(float *m, float *s, const float *l, const float *r, size_t count)
        {
            size_t off;

            __asm__ __volatile__
            (
                : [off] "=&r" (off),
                  [count] "+g" (count)
                : [left] "r"(l), [right] "r" (r),
                  [mid] "r" (m), [side] "r" (s)
                : "cc", "memory"
            );
        }
```

Here we see that there are five parameters that strictly require registers by using the `r` constraint and one general parameter defined by `g` constraint.
GCC parameter allocation works in such manner: it allocates 5 general-purpose registers for `[off]`, `[left]`, `[right]`, `[mid]` and `[side]` parameters and, after that, tries to allocate additional register for `[count]`. If there is no available register, then GCC decides to use memory instead.
According to CLANG's algorithm, it seems that `g` is allocated as a register without looking the count of `r` constraints. And that's why CLANG considers 6 registers being allocated (from total 8 available where ESP and EBP are busy for stack pointers) which is considered to be error at least because EBX is also busy for position-independent code purpose.
Changing constraints to this:

```
        void lr_to_ms(float *m, float *s, const float *l, const float *r, size_t count)
        {
            size_t off;

            __asm__ __volatile__
            (
                : [off] "=&r" (off),
                  [count] "m" (count)
                : [left] "r"(l), [right] "r" (r),
                  [mid] "r" (m), [side] "r" (s)
                : "cc", "memory"
            );
        }
```

Fixes problem.

I think, it is a good reason to report a but to clang's upstream. I'll try to provide the workaround.

**Comment by sadko4u** [comment]
Updated code to make it properly compile for i386 using CLANG compiler:
https://github.com/lsp-plugins/lsp-dsp-lib/commit/8ad15a86061cf55e4a679bd056184d03e3874943

**Comment by sadko4u** [comment]
Updated the code. Verified build on 32-bit ARM:
https://github.com/lsp-plugins/lsp-dsp-lib/commit/72bb6181d4b492c40ed196723db9e52e9447490b

**Comment by sadko4u** [comment]
Additionally, made possible build for AArch64.

**Comment by yurivict** [comment]
I'll try it when these changes get into release.

**Comment by sadko4u** [comment]
Available in 1.2.3 release!
